### PR TITLE
Feature/allow add emqtt proxy

### LINKED ISSUE #57: Allow admin to add eMQTT as proxy
Allow admins to add eMQTT broker through Proxies settings.

# User story
> As an APIKA administrator, 
I want to be able to add an eMQTT broker to APInf platform as an API proxy
so that API owners are able to add eMQTT APIs and connect them to the eMQTT.

# Definition of done
* [ ] When adding a proxy, user should be able to select between API Umbrella and eMQTT.
* [ ] If the user selects eMQTT, they should be able to fill in any required fields to configure the proxy, such as URL, ElasticSearch.

### PR DESCRIPTION
**Pull request by frenchbread**
Closes #57 

# Added
- File `proxies/collection/registered_proxies.js` for registering proxies (Needed for Proxies schema and `proxyForm` template)
- `eMQTT` schema
- Select box with proxies options (apiUmbrella, eMQTT)
- Import dependencies
- Check if proxy type is emqtt in `proxyItem` template & show related fields

# Changes
- Fixed `apiUmbrella` schema import in `proxies/collection/schemas/index.js`
- Cleanup `apiUmbrella` as default proxy type

### CODE REVIEW

**Comment by frenchbread** [comment]
@Digipalvelutehdas/apika-tiimi please review 👀

**Comment by marla-singer** [comment]
@frenchbread Reviewing

File: proxies/client/form/form.html [hunk 1]
```diff
@@ -16,17 +16,15 @@ <h4 class="modal-title" id="proxyFormModalLabel">
...
             {{> afQuickField name='name' }}
             {{> afQuickField name='description' }}
-            <!--
-            {{> afQuickField name='type' value='apiUmbrella' options="allowed" }}
-            -->
-            {{> afQuickField name='apiUmbrella' }}
-            <!--
-            {{# each proxyName in proxiesList }}
+
+            {{> afQuickField name='type' options="allowed" }}
```
Comment (marla-singer) [inline proxies/client/form/form.html]: Replace double quotes in single in word **allowed**

File: proxies/client/proxyItem/proxyItem.html [hunk 2]
```diff
@@ -56,6 +56,36 @@
           </tbody>
         </table>
       {{/ if }}
+      {{# if equals proxy.type 'emqtt' }}
+        <table class="table table-condensed">
+          <tbody>
+            <tr>
+              <th scope="row">
+                Type
```
Comment (marla-singer) [inline proxies/client/proxyItem/proxyItem.html]: Add i18n tag

File: proxies/client/proxyItem/proxyItem.html [hunk 3]
```diff
@@ -56,6 +56,36 @@
...
+              </th>
+              <td>
+                {{ proxy.type }}
+              </td>
+            </tr>
+            <tr>
+              <th scope="row">
+                URL
```
Comment (marla-singer) [inline proxies/client/proxyItem/proxyItem.html]: Add i18n tag

File: proxies/client/proxyItem/proxyItem.html [hunk 4]
```diff
@@ -56,6 +56,36 @@
...
+              </th>
+              <td>
+                {{ proxy.emqtt.url }}
+              </td>
+            </tr>
+            <tr>
+              <th scope="row">
+                Elasticsearch
```
Comment (marla-singer) [inline proxies/client/proxyItem/proxyItem.html]: Add i18n tag. Part for apiUmbrellatype has the same words. Please Add i18n tags for apiUmbrella type too

File: proxies/collection/schemas/emqtt.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  emqtt: {
+    type: Object,
+    optional: true,
+  },
+  'emqtt.url': {
+    type: String,
+    optional: false,
+    regEx: SimpleSchema.RegEx.Url,
+  },
+  // TODO: decide if these fields are needed for emqtt..
```
Comment (marla-singer) [inline proxies/collection/schemas/emqtt.js]: Delete comments. Issue has the needed fields
> If the user selects eMQTT, they should be able to fill in any required fields to configure the proxy, such as URL, ElasticSearch.

File: proxies/client/form/form.html (hunk 1 above)
Comment (frenchbread) [inline proxies/client/form/form.html]: Changing all to double quotes, so everything is consistent.

**Comment by frenchbread** [comment]
@marla-singer Done!

File: proxies/client/proxyItem/proxyItem.html [hunk 6]
```diff
@@ -15,39 +15,39 @@
           <tbody>
             <tr>
               <th scope="row">
-                Type
+                {{_ "proxyItem_apiUmbrella_type" }}
```
Comment (marla-singer) [inline proxies/client/proxyItem/proxyItem.html]: Change this tag name to correct: "proxyItem_apiUmbrella_proxyType"

**Comment by marla-singer** [comment]
@frenchbread Merge it

File: proxies/client/proxyItem/proxyItem.html [hunk 7]
```diff
@@ -15,7 +15,7 @@
           <tbody>
             <tr>
               <th scope="row">
-                {{_ "proxyItem_apiUmbrella_type" }}
+                {{_ "proxyItem_apiUmbrella_proxyType" }}
```
Comment (brylie) [inline proxies/client/proxyItem/proxyItem.html]: Should this be called `apiUmbrella_proxyType`? In other words, API Umbrella is a proxy type. Might it be called `proxyItem_proxyType`?

Comment (marla-singer) [inline proxies/client/proxyItem/proxyItem.html]: Below we have the table for eMQTT data which also has field "Type"

Comment (marla-singer) [inline proxies/client/proxyItem/proxyItem.html]: I renamed this tag because @frenchbread called so in en.i18n.json file and no more

Comment (frenchbread) [inline proxies/client/proxyItem/proxyItem.html]: @marla-singer OK, thanks.

**Pull request merged**